client/my-sites/plugins/main.jsx

```diff
@@ -142,8 +142,10 @@ export default React.createClass( {
 	},
 
 	getPluginsState( nextProps ) {
-		const sites = this.state && this.state.bulkManagement ? this.props.sites.getSelectedOrAllWithPlugins() : this.props.sites.getSelectedOrAll(),
-			pluginUpdate = PluginsStore.getPlugins( sites, 'updates' );
+		const sites = this.state && this.state.bulkManagement.length
```

Comment:
As it stands, `this.state.bulkMangement` shouldn't be anything but a list _unless_ it's specifically deleted since it's assigned in `getInitialState()` I would prefer to avoid checking the nested hierarchy for the sake of brevity. If it _could_ be undefined, then a good use of soft failure in destructuring will eliminate the error-checking for us.

```js
const { bulkManagement = [] } = this.state;
const { getSelectedOrAllWithPlugins, getSelectedOrAll } = this.props.sites;

const sites = bulkManagement.length
	? getSelectedOrAllWithPlugins()
	: getSelectedOrAll();
```